core-rust/core-api-server/core-api-schema.yaml

```diff
@@ -6965,6 +6965,12 @@ components:
       properties:
         network:
           $ref: "#/components/schemas/NetworkIdentifier"
+        at_state_version:
```

Comment:
The natural evolution would be to develop the dedicated `/find-state-version` endpoint first, and then consider the convenience field in the preview request (preferably - after anyone complains about the inconvenience).

> Although I don't know if we have good indices for round_timestamp => state_version at present?

We don't, but the timestamp is monotonic [or even ~linear?], so we can do the bisect [with very good guided jumps?] for log(n) responses. I'd honestly start this way rather than maintain an index.

Reply:
How about we make this an optional `at_ledger_state: { state_version: X }` [in accordance with the Gateway](https://radix-babylon-gateway-api.redoc.ly/#operation/EntityMetadataPage) whilst giving us options open to expand it in future?

Reply:
hmm, honestly I'd do it right away, but the schema used by Gateway in this case (i.e. a set of optional fields) is significantly different from our convention (polymorphism via discriminator).

Can I instead do `at_ledger_state?: { type: BY_STATE_VERSION, state_version: X }`? (purely for extendability sake)

Reply:
Yeah, that's a fair point! Happy for this approach.